catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImpl.java

```diff
@@ -431,6 +452,197 @@ public void init() {
+                            .before(metacard.getModifiedDate())) {
+                        remoteRegistryMetacards.put(registryId, metacard);
+                    }
+                } else {
+                    remoteRegistryMetacards.put(registryId, metacard);
+                }
+            }
+
+        } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+            String message = "Error querying for subscribed metacards;";
+            LOGGER.error("{} For source ids: {}", message, pollableSourceIds);
+            throw new FederationAdminException(message, e);
+        }
+
+        return remoteRegistryMetacards;
+    }
+
+    private Map<String, Metacard> getRegistryMetacardsMap() throws FederationAdminException {
+
+        Map<String, Metacard> registryMetacardsMap = new HashMap<>();
+        List<Metacard> registryMetacards = getRegistryMetacards();
+
+        for (Metacard metacard : registryMetacards) {
+            String registryId = metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                    .getValue()
+                    .toString();
+            registryMetacardsMap.put(registryId, metacard);
+        }
+
+        return registryMetacardsMap;
+    }
+
+    private void addOrUpdateMetacardsForRemoteRegistries() throws FederationAdminException {
+        addOrUpdateMetacardsForRemoteRegistries(pollableSourceIds);
+    }
+
+    private void addOrUpdateMetacardsForRemoteRegistries(Set<String> sourceIds)
+            throws FederationAdminException {
+        if (CollectionUtils.isEmpty(sourceIds)) {
+            throw new FederationAdminException(
+                    "Error adding or updating metacard for remote registry. Empty set of source ids provided.");
+        }
+
+        Map<String, Metacard> remoteRegistryMetacardsMap = getRemoteRegistryMetacardsMap();
+        Map<String, Metacard> registryMetacardsMap = getRegistryMetacardsMap();
+
+        Map<Serializable, Metacard> remoteMetacardsToUpdate = new HashMap<>();
+        List<Metacard> remoteMetacardsToCreate = new ArrayList<>();
+
+        for (Map.Entry<String, Metacard> remoteEntry : remoteRegistryMetacardsMap.entrySet()) {
+            if (registryMetacardsMap.containsKey(remoteEntry.getKey())) {
+                Metacard existingMetacard = registryMetacardsMap.get(remoteEntry.getKey());
+
+                Attribute localNode =
+                        existingMetacard.getAttribute(RegistryObjectMetacardType.REGISTRY_LOCAL_NODE);
+
+                if (localNode == null && remoteEntry.getValue()
+                        .getModifiedDate()
+                        .after(existingMetacard.getModifiedDate())) {
+                    remoteMetacardsToUpdate.put(remoteEntry.getKey(), remoteEntry.getValue());
+                }
+
+            } else {
+                remoteMetacardsToCreate.add(remoteEntry.getValue());
+            }
+        }
+
+        if (MapUtils.isNotEmpty(remoteMetacardsToUpdate)) {
```

Comment:
The CollectionUtils in apache.shiro works on collections and maps. I'm not using shiro for anything security so I'm using the CollectionUtils in apache.commons and that one doesn't work on Maps.